service_container/calls.rst

```diff
@@ -144,3 +141,21 @@ The configuration to tell the container it should do so would be like:
 
         $container->register(MessageGenerator::class)
             ->addMethodCall('withLogger', [new Reference('logger')], true);
+
+If autowire is enabled, you can also use annotations; with the previous exemple it would be:
+
+    .. code-block:: php
+
+        /**
+         * @required
+         * @return static
+         */
+        public function withLogger(LoggerInterface $logger)
+        {
+            $new = clone $this;
+            $new->logger = $logger;
+
+            return $new;
+        }
+
+You can also leverage the PHP8 `static` return type instead of the `@return static` annotation. Note if you don't want a method with a PHP8 `static` return type and a `@required` annotation to behave as a wither, you can add a `@return $this` annotation to disable the *returns clone* feature.
```

Comment:
Minor thing: in RST, code syntax is different than in Markdown. We must use two double backticks instead of one:

```rst
You can also leverage the PHP8 ``static`` return type instead of the ``@return static`` annotation [...]
```